### PR TITLE
Improve spectra selection within Sliceviewer downsampling

### LINKED ISSUE #29237: Sliceviewer is sampling, and not rebinning
**Original reporter:** Tatiana Guidi

### Expected behavior
If there is a high counting detector it should be visible in some way on the initial view of the workspace.
So the scientists know it is there, diagnostics etc.

Spectrum viewer used to show these as it rebinned the underlying data, so a single outlying spectra was averaged, not ignored.

### Actual behavior
Due to the sampling approach we are using, it is perfectly possible for a single high counting detector to not be shown at all on the initial view, and only become visible when zooming in.

### Steps to reproduce the behavior
An example of this is Sp 795 on MAR27283.

You can't see it on the initial slice viewer view, but only if you maximise the window, or zoom in, and with a larger instrument it will happen more often and the maximising approach will not work as often.

### Platforms affected
All

Mentioning a couple of people so they know about this one:
@martyngigg @AnthonyLim23

### PR DESCRIPTION
**Pull request by StephenSmith25**
**Description of work.**

This PR improves the spectra selection within the sliceviewer when downsampling the image. This avoids missing high counting detectors when the sliceviewer is first opened. To solve this issue, a maxpooling approach is used, which ensures that within each sampling range (ymin, ymax) the highest counting detector is selected.

**To test:**

1. Code review
2. As per the attatched issue, do the following:
3. Load MAR27283.nxs 
4. Open sliceviewer on the workspace (**note**: on Debug builds this appears to take a while...)
5. The high counting spectra at 794 should now be shown (previously it wasn't)

Fixes #29237 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by robertapplin — CHANGES_REQUESTED** [review]
Code changes look good and good tests 👍 Just have one comment/question

File: Framework/PythonInterface/mantid/plots/resampling_image/samplingimage.py [hunk 1]
```diff
@@ -138,6 +142,17 @@ def _update_extent(self):
     def get_full_extent(self):
         return self._full_extent
 
+    def _update_maxpooling_option(self):
+        """
+        Updates the maxpooling option, used when the image is downsampled
+        If the workspace is large, we skip this maxpooling step and set the option as False
+        """
+        axis = self.ws.getAxis(1)
+        if self.ws.getNumberHistograms() < MAX_HISTOGRAMS and axis.isSpectra():
```
[the remark this answers is not in the document]
Comment (StephenSmith25) [reply Framework/PythonInterface/mantid/plots/resampling_image/samplingimage.py:151]: Yeah good point, I guess from the name `MAX_HISTOGRAMS` you'd probably expect the `<=` operator. So I agree, this should be changed 👍

**Comment by robertapplin** [comment]
Looks like there is a merge conflict @StephenSmith25

**Review by robertapplin — APPROVED** [review]
Looks good, test failure is unrelated.

**Pull request merged**